examples/whisper/run.py

```diff
@@ -163,42 +163,50 @@ def get_session(self, engine_dir, runtime_mapping, debug_mode=False):
 
         return decoder_generation_session
 
-    def generate(self,
-                 decoder_input_ids,
-                 encoder_outputs,
-                 eot_id,
-                 max_new_tokens=40,
-                 num_beams=1):
+    def generate(
+        self, decoder_input_ids, encoder_outputs, eot_id, max_new_tokens=40, num_beams=1
+    ):
+
+        if torch.is_tensor(encoder_outputs):
+            encoder_outputs = list(encoder_outputs)
+        if torch.is_tensor(decoder_input_ids):
+            decoder_input_ids = list(decoder_input_ids)
+
         encoder_input_lengths = torch.tensor(
-            [encoder_outputs.shape[1] for x in range(encoder_outputs.shape[0])],
+            [encoder_output.shape[0] for encoder_output in encoder_outputs],
             dtype=torch.int32,
-            device='cuda')
-        decoder_input_lengths = torch.tensor([
-            decoder_input_ids.shape[-1]
-            for _ in range(decoder_input_ids.shape[0])
-        ],
-                                             dtype=torch.int32,
-                                             device='cuda')
+            device="cuda",
+        )
+        decoder_input_lengths = torch.tensor(
+            [decoder_input_id.shape[-1] for decoder_input_id in decoder_input_ids],
+            dtype=torch.int32,
+            device="cuda",
+        )
         decoder_max_input_length = torch.max(decoder_input_lengths).item()
 
-        cross_attention_mask = torch.ones(
-            [encoder_outputs.shape[0], 1,
-             encoder_outputs.shape[1]]).int().cuda()
+        cross_attention_mask = (
```

Comment:
Hi @MahmoudAshraf97 if I understand correctly, you are making this change because distil-whisper can work on dynamic chunk sizes, unlike whisper which must use fixed 30 second chunks. Am I understanding correctly? Thank you.

Reply:
Hi @galv , this PR contains 2 main changes
1. Encoder is no longer restricted to 30s inputs, this helps in case of distil-whisper as you mentioned
2. Decoder now supports `remove_input_padding` and accepts packed inputs to save memory